### PR TITLE
Make BaseJob.most_recent_job favor "running" jobs

### LINKED ISSUE #28118: Scheduler heartbeat warning message in Airflow UI displaying that scheduler is down sometimes incorrect
### Apache Airflow version

main (development)

### What happened

Steps to reproduce:
1. run 2 replicas of scheduler
2. initiate shut down of one of the schedulers
3. In Airflow UI observe message
image

3rd step should be done immediately after 2nd (refreshing UI page few times). 2nd and 3rd steps might be repeated for couple of times in order to reproduce.

### What you think should happen instead

Warning message shouldn't be displayed.

The issue is that for this warning message recent (with latest heartbet) scheduler job is fetched
https://github.com/apache/airflow/blob/f02a7e9a8292909b369daae6d573f58deed04440/airflow/jobs/base_job.py#L133.

And this may point to job which is not running (state!="running") and that is why we see warning message.
The warning message in this case is misleading as another replica of scheduler is running in parallel.

### How to reproduce

_No response_

### Operating System

Linux

### Versions of Apache Airflow Providers

_No response_

### Deployment

Composer

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kosteev**
Closes: #28118

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: airflow/jobs/base_job.py [hunk 1]
```diff
@@ -125,12 +125,23 @@ def most_recent_job(cls, session=None) -> BaseJob | None:
...
         This method should be called on a subclass (i.e. on SchedulerJob) to
         return jobs of that type.
 
         :param session: Database session
         """
-        return session.query(cls).order_by(cls.latest_heartbeat.desc()).limit(1).first()
+        return (
+            session.query(cls)
+            .order_by(
+                # Make sure "running" jobs first in order.
```
Comment (uranusjr) [inline airflow/jobs/base_job.py]:
```suggestion
                # Put "running" jobs at the front.
```

**Review by uranusjr — DISMISSED** [review]

**Comment by uranusjr** [comment]
MSSQL does not support ordering with a filter expression 😞 You probably need to use `COALESCE` or `ISNULL` instead and observe various databases support.

**Review by uranusjr — APPROVED** [review]
Assuming tests pass

**Comment by kosteev** [comment]
I see that MsSQL tests pass but Sqlite tests for some reason fail and I do not see logs here:
https://github.com/apache/airflow/actions/runs/3627891927/jobs/6118344061

Can you, please, help me out to understand why they are failing?

Also, should I do squash and rebase or it will be done automatically on PR merge?

**Comment by uranusjr** [comment]
This happens from time to time due to resource restrictions. I’ve restarted the job.

**Comment by kosteev** [comment]
Thanks

**Comment by kosteev** [comment]
It failed again, maybe retry one more time?

**Comment by potiuk** [comment]
> @uranusjr Can you, please, rerun it. It failed again.

I did. BTW. See devlist  - I will work on improving those flaky tests..

**Comment by kosteev** [comment]
Thx

**Pull request merged**